pkgs/misc/cups/libcupsfilters.nix

```
@@ -0,0 +1,102 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, avahi
+, cups
+, dejavu_fonts
+, dbus
+, freetype
+, fontconfig
+, ghostscript
+, lcms2
+, libexif
+, libjpeg
+, libpng
+, libtiff
+, pkg-config
+, poppler_utils
+, mupdf # mutool
+, poppler
+, qpdf
+, withAvahi ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libcupsfilters";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libcupsfilters";
+    rev = "7b6acace357b01129690849469496d52c98a6793";
+    hash = "sha256-kFjGYcj4kjFwD6q9Oa+fVvd2lTSQ6PUZWwg4oG3tKo8=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    libexif
+    libjpeg
+    libpng
+    libtiff
+    lcms2
+    freetype
+    fontconfig
+    qpdf
+    poppler
+    dbus
+    ghostscript
+    mupdf.bin
+  ] ++ lib.optionals withAvahi [ avahi ];
+
+  configureFlags = [
+    "--with-shell=${stdenv.shell}"
+    "--with-test-font-path=${dejavu_fonts}/share/fonts/truetype/DejaVuSans.ttf"
+    "--localstatedir=/var"
+    "--sysconfdir=/etc"
+  ] ++ lib.optionals (!withAvahi) [ "--disable-avahi" ];
+
+  prePatch = ''
+    # Patch upstream syntax error when !HAVE_OPEN_MEMSTREAM (on darwin)
+    substituteInPlace cupsfilters/bannertopdf.c \
+      --replace \
+      'log(ld, CF_LOGLEVEL_ERROR, "cfFilterBannerToPDF: Cannot malloc(): %s\n", , strerror(errno));' \
+      'log(ld, CF_LOGLEVEL_ERROR, "cfFilterBannerToPDF: Cannot malloc(): %s\n", strerror(errno));' \
+      --replace \
+      'log(ld, CF_LOGLEVEL_ERROR, "cfFilterBannerToPDF: Cannot fstat(): %s\n", , strerror(errno));' \
+      'log(ld, CF_LOGLEVEL_ERROR, "cfFilterBannerToPDF: Cannot fstat(): %s\n", strerror(errno));'
+
+    # execvpe is a GNU extension
+    ${lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc")
+      ''
+        substituteInPlace cupsfilters/ghostscript.c \
+          --replace \
+          'execvpe(filename, gsargv, envp);' \
+          'extern char **environ; environ = envp; execvp(filename, gsargv);'
+      ''
+    }
+    '';
+
+  makeFlags = [ "CUPS_SERVERBIN=$(out)/lib/cups" "CUPS_DATADIR=$(out)/share/cups" "CUPS_SERVERROOT=$(out)/etc/cups" ];
+
+  postConfigure =
+    ''
```

```suggestion
  makeFlags = [
    "CUPS_SERVERBIN=$(out)/lib/cups"
    "CUPS_DATADIR=$(out)/share/cups"
    "CUPS_SERVERROOT=$(out)/etc/cups"
  ];

  postConfigure = ''
```
normally we do one list element per line for clarity and to get better diffs.

---

resolved in rebased commits